From 47b1d81b04a334b8115654f0600631373c250e1d Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Wed, 16 Dec 2015 18:00:30 +0800
Subject: [PATCH 0775/5242] MLK-12029 ARM: dts: imx6qp-sabreauto.dts: add vmmc
 supply for usdhc3

commit  e4faa76353eb59d300453e010001049ef1e742e8 from
https://source.codeaurora.org/external/imx/linux-imx.git

For imx6qp-ard board, it is okay to use external regulator for card slot,
so this patch add vmmc-supply for usdhc3.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |   10 ++++++++++
 arch/arm/boot/dts/imx6qp-sabreauto.dts   |    8 ++++++++
 2 files changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index cf3d50d..7dfe85b 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -155,6 +155,16 @@
 			regulator-max-microvolt = <5000000>;
 			regulator-always-on;
 		};
+
+		reg_sd3_vmmc: regulator@7 {
+			compatible = "regulator-fixed";
+			regulator-name = "P3V3_SDa_SWITCHED";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio7 8 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+			status = "disabled";
+		};
 	};
 
 	hannstar_cabc {
diff --git a/arch/arm/boot/dts/imx6qp-sabreauto.dts b/arch/arm/boot/dts/imx6qp-sabreauto.dts
index 3e8ac46..4bee45f 100644
--- a/arch/arm/boot/dts/imx6qp-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6qp-sabreauto.dts
@@ -72,10 +72,18 @@
 	status = "okay";
 };
 
+&reg_sd3_vmmc {
+	status = "okay";
+};
+
 &sata {
 	status = "okay";
 };
 
+&usdhc3 {
+	vmmc-supply = <&reg_sd3_vmmc>;
+};
+
 &vgen3_reg {
 	regulator-always-on;
 };
-- 
1.7.9.5

